addons/binding/org.openhab.binding.keba/ESH-INF/thing/kecontactp20.xml

```diff
@@ -0,0 +1,132 @@
+        xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+        xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+    <!-- Sample Thing Type -->
+    <thing-type id="kecontactp20">
+        <label>Keba KeContact P20</label>
+        <description>This Thing supports the KeContact P20 EV Charging station</description>
+    
+        <channels>
+            <channel id="model" typeId="model"/>
+            <channel id="firmware" typeId="firmware"/>
+            <channel id="state" typeId="state"/>
+            <channel id="error" typeId="error"/>
+            <channel id="wallbox" typeId="plug"/>
+            <channel id="vehicule" typeId="plug"/>
+            <channel id="locked" typeId="locked"/>
+            <channel id="enabled" typeId="switch"/>
+            <channel id="maxsystemcurrent" typeId="current"/>
+            <channel id="maxpresetcurrentrange" typeId="range"/>
+            <channel id="maxpresetcurrent" typeId="current"/>
+            <channel id="failsafecurrent" typeId="current"/>
+            <channel id="output" typeId="switch"/>
+            <channel id="input" typeId="switch"/>
+            <channel id="serial" typeId="serial"/>
+            <channel id="uptime" typeId="uptime"/>
+            <channel id="I1" typeId="current"/>
+            <channel id="I2" typeId="current"/>
+            <channel id="I3" typeId="current"/>
+            <channel id="U1" typeId="voltage"/>
+            <channel id="U2" typeId="voltage"/>
+            <channel id="U3" typeId="voltage"/>
+            <channel id="power" typeId="power"/>
+            <channel id="powerfactor" typeId="powerfactor"/>
+            <channel id="sessionconsumption" typeId="energy"/>
+            <channel id="totalconsumption" typeId="energy"/>
+        </channels>
+            
+         <config-description>
+            <parameter name="ipAddress" type="text" required="true">
+                <label>Network Address</label>
+                <description>Network address of the KEBA KeContact P20</description>
+            </parameter>
+            <parameter name="refreshInterval" type="integer" required="false">
+            	<label>Refresh interval</label>
+            	<description>Specifies the refresh interval in seconds.</description>
+            	<default>15</default>
+        	</parameter>      
+        </config-description>
+
+    </thing-type>
+
+    <!-- Sample Channel Type -->
+    <channel-type id="model">
+        <item-type>String</item-type>
+        <label>Model</label>
+        <description>KeContact P20 Model/Variant</description>
+    </channel-type>
+    <channel-type id="firmware">
+        <item-type>String</item-type>
+        <label>Firmware</label>
+        <description>Firmware version number</description>
+    </channel-type>
+    <channel-type id="state">
```

Comment:
Please also make use of additional meta-data throughout this file: Use advanced, read-only, pattern, options etc.